plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/internal/EclipseBasedN4JSWorkspace.java

```diff
@@ -83,6 +83,16 @@ public URI findProjectWith(URI nestedLocation) {
 				&& nestedLocation.segmentCount() >= DIRECT_RESOURCE_IN_PROJECT_SEGMENTCOUNT) {
 			return URI.createPlatformResourceURI(nestedLocation.segment(1), true);
 		}
+		if (nestedLocation.isFile()) {
```

Comment:
Why would file in eclipse project not be a platform resource? 
In which cases is this fallback needed?

Reply:
In case of a source map, the file is retrieved via the absolute or relative path in the source map.